asmcomp/cmmgen.ml

```diff
@@ -2317,6 +2319,20 @@ and transl_prim_2 env p arg1 arg2 dbg =
 
 and transl_prim_3 env p arg1 arg2 arg3 dbg =
   match p with
+  (* Heap operations *)
+  | Psetfield_computed(ptr, init) ->
+      begin match init, ptr with
+      | (Assignment | Heap_initialization), Pointer ->
+        return_unit (
+          addr_array_set (transl env arg1) (transl env arg2) (transl env arg3)
+            dbg)
```

Comment:
Is there a reason not to use `caml_initialize` instead of `caml_modify` in case of initialization with a pointer value ?
If this is an error, to avoid potential future problems, I also suggest sharing the logic by something like:
```OCaml
type assignment_kind = Caml_modify | Caml_initialize | Simple

let assignment_kind ptr init =
  match init, ptr with
  | Assignment, Pointer -> Caml_modify
  | Heap_initialization, Pointer -> Caml_initialize
  | Assignment, Immediate
  | Heap_initialization, Immediate
  | Root_initialization, (Immediate | Pointer) -> Simple
```